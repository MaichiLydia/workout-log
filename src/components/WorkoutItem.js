import React, { Component } from 'react'
import { faTimes } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import moment from 'moment'
import PropTypes from "prop-types";

class WorkoutItem extends Component {
  
  render() {
    return (
      <tr className="table__content-row">
        <td className="table__content-column"> {this.props.workout.duration}</td>
        <td className="table__content-column"> {this.props.activity}</td>
        <td className="table__content-column"> {moment(this.props.workout.date).format('DD/MM/YYYY')}</td>
        <td className="table__content-column"> <button className="table__btn" title="Remove item" onClick={() => this.props.removeWorkout(this.props.workout.id)}><FontAwesomeIcon icon={faTimes} /></button>  
        </td>
      </tr>
    )
  }
}

WorkoutItem.propTypes = {
  workout: PropTypes.object,
  removeWorkout: PropTypes.func,
};

export default WorkoutItem
