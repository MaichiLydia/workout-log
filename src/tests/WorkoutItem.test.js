import React from "react";
import WorkoutItem from "../components/WorkoutItem"
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';


it('check if WorkoutItem receives the right values', () => {
  const props = {
        workout:{
          id: "1",
          duration:"10:10",
          "activityId": 2,
          "date": '2019-01-01'
        }
      }
  const select = mount(<WorkoutItem {...props}/>)
  expect(toJson(select)).toMatchSnapshot()
});
