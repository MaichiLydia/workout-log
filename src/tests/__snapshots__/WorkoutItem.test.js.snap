// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if WorkoutItem receives the right values 1`] = `
<WorkoutItem
  workout={
    Object {
      "activityId": 2,
      "date": "2019-01-01",
      "duration": "10:10",
      "id": "1",
    }
  }
>
  <tr
    className="table__content-row"
  >
    <td
      className="table__content-column"
    >
       
      10:10
    </td>
    <td
      className="table__content-column"
    >
       
    </td>
    <td
      className="table__content-column"
    >
       
      01/01/2019
    </td>
    <td
      className="table__content-column"
    >
       
      <button
        className="table__btn"
        onClick={[Function]}
        title="Remove item"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                352,
                512,
                Array [],
                "f00d",
                "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
              ],
              "iconName": "times",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-times fa-w-11 "
            data-icon="times"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 352 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </FontAwesomeIcon>
      </button>
    </td>
  </tr>
</WorkoutItem>
`;
